build_and_test.sh

```diff
@@ -112,28 +78,23 @@ fi
 
 #Build and run test with shared memory model
 cd $SHM_BUILD_DIR
- cmake ../.. -DARG_OPENFAM_MODEL=shared_memory; make ; make install
+ cmake ../.. -DARG_OPENFAM_MODEL=shared_memory; $MAKE_CMD ; make install
```

Comment:
You have submitted code changes in all test cases to remove SHM build type, 
I think we should have only one build for both memory_server and shared_memory model.

Similar changes required in CMakeLists.txt file and remove the usage of this argument -DARG_OPENFAM_MODEL=shared_memory